parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -67,10 +68,29 @@ $global:AgentCertificate = "{{GetParameter "clientCertificate"}}"
 $global:KubeBinariesPackageSASURL = "{{GetParameter "kubeBinariesSASURL"}}"
 $global:WindowsKubeBinariesURL = "{{GetParameter "windowsKubeBinariesURL"}}"
 $global:KubeBinariesVersion = "{{GetParameter "kubeBinariesVersion"}}"
+$global:ContainerdUrl = "{{GetParameter "windowsContainerdURL"}}"
+$global:ContainerdSdnPluginUrl = "{{GetParameter "windowsSdnPluginURL"}}"
 
 ## Docker Version
 $global:DockerVersion = "{{GetParameter "windowsDockerVersion"}}"
 
+## ContainerD Usage
+$global:ContainerRuntime = "{{GetParameter "containerRuntime"}}"
+$global:DefaultContainerdRuntimeHandler = "{{GetParameter "defaultContainerdRuntimeHandler"}}"
+$global:HypervRuntimeHandlers = "{{GetParameter "hypervRuntimeHandlers"}}"
+
+# To support newer Windows OS version, we need to support set ContainerRuntime,
+# HypervRuntimeHandlers and DefaultContainerdRuntimeHandler per agent pool but
+# current code does not support this. Below is a workaround to set contianer
+# runtime variables per Windows OS version.
+#
+# Set default values for container runtime variables for AKS Windows 2004
+if ($([System.Environment]::OSVersion.Version).Build -eq "19041") {

```

Comment:
There are below issues in this file which do not match the code in aks-engine:
1. `$global:EnableHostsConfigAgent = [System.Convert]::ToBoolean("{{ EnableHostsConfigAgent }}");` does not match `$global:EnableHostsConfigAgent = [System.Convert]::ToBoolean("{{WrapAsVariable "enableHostsConfigAgent" }}");`
1. 
```
. c:\AzureData\windows\kuberneteswindowsfunctions.ps1
. c:\AzureData\windows\windowsconfigfunc.ps1
. c:\AzureData\windows\windowskubeletfunc.ps1
. c:\AzureData\windows\windowscnifunc.ps1
. c:\AzureData\windows\windowsazurecnifunc.ps1
. c:\AzureData\windows\windowscsiproxyfunc.ps1
. c:\AzureData\windows\windowsinstallopensshfunc.ps1
. c:\AzureData\windows\windowscontainerdfunc.ps1
. c:\AzureData\windows\windowshostsconfigagentfunc.ps1
```
does not match 
```
. c:\AzureData\k8s\kuberneteswindowsfunctions.ps1
. c:\AzureData\k8s\windowsconfigfunc.ps1
. c:\AzureData\k8s\windowskubeletfunc.ps1
. c:\AzureData\k8s\windowscnifunc.ps1
. c:\AzureData\k8s\windowsazurecnifunc.ps1
. c:\AzureData\k8s\windowscsiproxyfunc.ps1
. c:\AzureData\k8s\windowsinstallopensshfunc.ps1
. c:\AzureData\k8s\windowscontainerdfunc.ps1
. c:\AzureData\k8s\windowshostsconfigagentfunc.ps1
```
1. `-AADClientSecret $AADClientSecret ` does not match `-AADClientSecret $([System.Text.Encoding]::ASCII.GetString([System.Convert]::FromBase64String($AADClientSecret)))`
1. `-TargetEnvironment $TargetEnvironment` does not match `-TargetEnvironment {{if IsAKSCustomCloud}}"AzureStackCloud"{{else}}$TargetEnvironment{{end}}`
1. There is no 
```
        {{if or IsAzureStackCloud IsAKSCustomCloud}}
        $azureStackConfigFile = [io.path]::Combine($global:KubeDir, "azurestackcloud.json")
        $envJSON = "{{ GetBase64EncodedEnvironmentJSON }}"
        [io.file]::WriteAllBytes($azureStackConfigFile, [System.Convert]::FromBase64String($envJSON))
        {{end}}
```
and
```
{{- /* Azure Stack has discrete Azure CNI config requirements */}}
                -IsAzureStack {{if IsAzureStackCloud}}$true{{else}}$false{{end}} `
```

Reply:
>. $global:EnableHostsConfigAgent = [System.Convert]::ToBoolean("{{ EnableHostsConfigAgent }}"); does not match $global:EnableHostsConfigAgent = [System.Convert]::ToBoolean("{{WrapAsVariable "enableHostsConfigAgent" }}");

I made a change to use function `EnableHostsConfigAgent` defined [here](https://github.com/Azure/AgentBaker/blob/master/pkg/agent/baker.go#L339), which is shared with Linux Cloudinit generation

2. we changed the directory of Windows related custom data from `k8s` to `Windows`

3. > -AADClientSecret $AADClientSecret does not match -AADClientSecret $([System.Text.Encoding]::ASCII.GetString([System.Convert]::FromBase64String($AADClientSecret)))

I have an explanation for this change in https://github.com/Azure/AgentBaker/pull/396#discussion_r512040860.  We do not encode `AADClientSecret`, so we don't need to decode it.

4. > -TargetEnvironment $TargetEnvironment does not match -TargetEnvironment {{if IsAKSCustomCloud}}"AzureStackCloud"{{else}}$TargetEnvironment{{end}}

We don't support `AzureStackCloud`, deleting this for simplicity

5. As we don't support AzureStackCloud, so I just removed `IsAzureStackCloud` function for simplicity